packages/element/src/react.js

```diff
@@ -133,3 +133,24 @@ export function switchChildrenNodeName( children, nodeName ) {
 		return createElement( nodeName, { key: index, ...props }, childrenProp );
 	} );
 }
+
+/**
+ * Function that returns an array of all the text elements
+ * descendants of the element passed.
+ *
+ * @param {?Array|string|WPElement} element Element object, a string representing a text node
+ *                                          or an array of nodes.
+ *
+ * @return {?Array} An array of all text nodes present on the tree of the element passed.
+ */
+export function getTextElements( element ) {
+	if ( isString( element ) ) {
+		return element;
+	}
+	if ( isArray( element ) ) {
+		return flatMap( element, ( child ) => getTextElements( child ) );
+	}
+	if ( element && element.props && element.props.children ) {
```

Comment:
Isn't there some lodash helper for this kind of deep-check? `get()` or something?

I personally don't like it and prefer just using "raw" JS, but we use it elsewhere so we might as well here 🤷‍♂️

Reply:
Bring on https://github.com/tc39/proposal-optional-chaining! 🙂